### PR TITLE
Use default system browser

### PR DESCRIPTION
**Pull request by taooceros**
#203 Use the Default Browsers in the computer instead of simply using chrome as default.

### CODE REVIEW

File: Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs [hunk 1]
```diff
@@ -1,18 +1,45 @@
...
 namespace Flow.Launcher.Plugin.SharedCommands
 {
     public static class SearchWeb
     {
+        private static string GetDefaultBrowserPath()
+        {
+            string name = string.Empty;
+            try
+            {
+                var regDefault = Registry.CurrentUser.OpenSubKey("Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\FileExts\\.htm\\UserChoice", false);
```
Comment (JohnTheGr8) [inline Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs:17]: why
`Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\FileExts\\.htm\\UserChoice`
and not
`Software\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\http\\UserChoice`
?

File: Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs [hunk 2]
```diff
@@ -1,18 +1,45 @@
...
+                var stringDefault = regDefault.GetValue("ProgId");
+
+                using var regKey = Registry.ClassesRoot.OpenSubKey(stringDefault + "\\shell\\open\\command", false);
+                name = regKey.GetValue(null).ToString().ToLower().Replace("" + (char)34, "");
```
Comment (JohnTheGr8) [inline Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs:21]: This can be simplified to:
```csharp
name = regKey.GetValue("").ToString().Replace("\"", "");
```

File: Flow.Launcher/SettingWindow.xaml.cs [hunk 3]
```diff
@@ -263,7 +263,7 @@ private async void OnCheckUpdates(object sender, RoutedEventArgs e)
 
         private void OnRequestNavigate(object sender, RequestNavigateEventArgs e)
         {
-            SearchWeb.NewBrowserWindow(e.Uri.AbsoluteUri);
+            SearchWeb.NewTabInBrowser(e.Uri.AbsoluteUri);
```
Comment (JohnTheGr8) [inline Flow.Launcher/SettingWindow.xaml.cs:266]: Can you also replace the use of `SearchWeb.NewBrowserWindow` in other places in settings?

File: Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs (hunk 1 above)
Comment (JohnTheGr8) [reply Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs:17]: pleas also add a `using` here

**Review by JohnTheGr8 — CHANGES_REQUESTED** [review]

File: Flow.Launcher/SettingWindow.xaml.cs (hunk 3 above)
Comment (taooceros) [reply Flow.Launcher/SettingWindow.xaml.cs:266]: Sure.

File: Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs (hunk 1 above)
Comment (taooceros) [reply Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs:17]: OK, I am not sure about the use of registry since I am not familier with it, so I just copy one and it seems it works. I will test the one you suggested and it seems it will be more reasonable to use that one instead of the htm default link.

File: Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs (hunk 2 above)
Comment (taooceros) [reply Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs:21]: Thanks for suggestion.

File: Flow.Launcher/ReportWindow.xaml.cs [hunk 4]
```diff
@@ -52,8 +52,8 @@ private Paragraph Hyperlink(string textBeforeUrl, string url)
             var link = new Hyperlink { IsEnabled = true };
             link.Inlines.Add(url);
             link.NavigateUri = new Uri(url);
-            link.RequestNavigate += (s, e) => SearchWeb.NewBrowserWindow(e.Uri.ToString());
-            link.Click += (s, e) => SearchWeb.NewBrowserWindow(url);
+            link.RequestNavigate += (s, e) => SearchWeb.NewTabInBrowser(e.Uri.ToString());
+            link.Click += (s, e) => SearchWeb.NewTabInBrowser(url);
```
Comment (jjw24) [inline Flow.Launcher/ReportWindow.xaml.cs:56]: why is this changed to open as a tab?

File: Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs [hunk 5]
```diff
@@ -1,18 +1,45 @@
...
+            return name;
+        }
+
         /// <summary> 
         /// Opens search in a new browser. If no browser path is passed in then Chrome is used. 
         /// Leave browser path blank to use Chrome.
         /// </summary>
 		public static void NewBrowserWindow(this string url, string browserPath = "")
         {
+            browserPath = string.IsNullOrEmpty(browserPath) ? GetDefaultBrowserPath() : browserPath;
```
Comment (jjw24) [inline Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs:41]: with getting the default browser from user, which could be any browser right, do we know if it will support opening in new window?

**Comment by jjw24** [comment]
Why are we changing the behaviour of opening hyperlinks in Flow from new window to as a tab? I think the core of the issue is supporting default browser right?

File: Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs (hunk 5 above)
Comment (JohnTheGr8) [reply Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs:41]: It works on firefox and chromium-based browsers, and there is a fix for IE a few lines later.... I think this covers pretty much every browser.

**Review by JohnTheGr8 — COMMENTED** [review]
> 
> 
> Why are we changing the behaviour of opening hyperlinks in Flow from as a new window to as a tab? I think the core of the issue is supporting default browser right?

opening in a new tab is expected behavior I think, no app opens links in a new window AFAIK...

File: Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs [hunk 6]
```diff
@@ -1,18 +1,45 @@
+using Microsoft.Win32;
 using System;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
 
+
```
Comment (jjw24) [inline Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs:7]: lets remove this spacing

File: Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs [hunk 7]
```diff
@@ -44,7 +71,10 @@ public static void NewBrowserWindow(this string url, string browserPath = "")
         /// </summary>
         public static void NewTabInBrowser(this string url, string browserPath = "")
         {
-            var psi = new ProcessStartInfo() { UseShellExecute = true};
+            browserPath = string.IsNullOrEmpty(browserPath) ? GetDefaultBrowserPath() : browserPath;
+
```
Comment (jjw24) [inline Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs:75]: remove spacing please

**Comment by taooceros** [comment]
Yes, and also we didn't provide the option for user to choose open in new window or tap. Besides, the default system behavior for browsers except is opening websites in new tab.

**Comment by jjw24** [comment]
Looks good to me. Good to go from you @JohnTheGr8 ?

**Comment by jjw24** [comment]
I will add Flow.Launcher.Plugin version bump to the 1.5.0 version bump pr once this goes in.

**Review by JohnTheGr8 — APPROVED** [review]
